### PR TITLE
[Permissions] Data permissions graph utils reorganization

### PR DESCRIPTION
**Pull request by sloansparger**
### Description

`frontend/src/metabase/admin/permissions/utils/graph/data-permissions.ts` has been increasingly hard to navigate with ~1700 LoC. This PR splits up the code into three major areas of concern:
1. accessing parts of the data permissions graph
2. updating the graph
3. detecting permissions existing in the graph

This is part of a iterative effort of mine to take our permissions utils and organize it into a library with a simpler (and documented) API and make it more easily unit tested.

### How to verify

The code should be functionally equivalent to what existed before.

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR
Not needed

### CODE REVIEW

File: frontend/src/metabase/admin/permissions/selectors/data-permissions/diff.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { createSelector } from "@reduxjs/toolkit";
 import _ from "underscore";
 
-import { diffDataPermissions } from "metabase/admin/permissions/utils/graph";
+import { diffDataPermissions } from "metabase/admin/permissions/utils/graph/permissions-diff";
```
Comment (iethree) [inline frontend/src/metabase/admin/permissions/selectors/data-permissions/diff.ts:4]: why this change? that index file is still intact.

**Review by iethree — APPROVED** [review]

File: frontend/src/metabase/admin/permissions/selectors/data-permissions/diff.ts (hunk 1 above)
Comment (sloansparger) [reply frontend/src/metabase/admin/permissions/selectors/data-permissions/diff.ts:4]: Ah I think I moved this then moved it back. I guess I didn't revert it correctly. Will fix.
